$OpenBSD: patch-src_main_c,v 1.23 2012/12/19 15:04:36 dcoppa Exp $

Fix some problems on initial volume setup
(upstream svn revision r2378)

--- src/main.c.orig	Wed Dec 19 13:36:07 2012
+++ src/main.c	Wed Dec 19 15:23:05 2012
@@ -1288,7 +1288,6 @@ int main(int argc, char *argv[])
     gm_audio_update_device(&audio_device);
     // disabling this line seems to help with hangs on startup when using pulseaudio
     //gm_audio_get_volume(&audio_device);
-    gm_audio_set_server_volume_update_callback(&audio_device, set_volume);
     set_media_player_attributes(media);
 
     if (!softvol) {
@@ -1362,6 +1361,7 @@ int main(int argc, char *argv[])
     safe_to_save_default_playlist = TRUE;
 
     // put the request to update the volume into the list of tasks to complete
+    g_idle_add(hookup_volume, NULL);
     g_idle_add(set_volume, NULL);
     gtk_main();
 
